### PR TITLE
[Refactor]: Removed SessionManager Async Creation Method

### PR DESCRIPTION
**Pull request by marcomaddy1982**
## What's new in this PR?

According to the [RFC](https://wearezeta.atlassian.net/wiki/spaces/IOS/pages/159646019/Simplify+API+for+Login+Registration) about the login and registration API one of the improvement we would like to add is removing the async creation of the `SessionManager` in the `AppRootRouter`. To achieve this goal some changes are made in the `SessionManager`:

- remove the method:

```swift
public static func create(
        appVersion: String,
        mediaManager: MediaManagerType,
        analytics: AnalyticsType?,
        delegate: SessionManagerDelegate?,
        presentationDelegate: PresentationDelegate?,
        application: ZMApplication,
        environment: BackendEnvironmentProvider,
        configuration: SessionManagerConfiguration,
        detector: JailbreakDetectorProtocol = JailbreakDetector(),
        completion: @escaping (SessionManager) -> Void
```

- expose the `pubic init` method 
- remove the logic that wait the protected data available to create the SessionManager (it will be moved in the UI).

## Dependencies

[JIRA](https://wearezeta.atlassian.net/browse/SQCORE-285)

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]

File: Source/SessionManager/SessionManager.swift [hunk 1]
```diff
@@ -389,14 +353,14 @@ public final class SessionManager : NSObject, SessionManagerType {
         self.dispatchGroup = dispatchGroup
         self.configuration = configuration.copy() as! SessionManagerConfiguration
         self.jailbreakDetector = detector
-
+
```
Comment (billypchan) [inline Source/SessionManager/SessionManager.swift:356]: I would suggest running my lint script to undo this indent space changes:

https://github.com/billypchan/iosGitHubTools/blob/master/scripts/lint.sh

**Review by billypchan — COMMENTED** [review]
undo indent space change

File: Source/SessionManager/SessionManager.swift (hunk 1 above)
Comment (marcomaddy1982) [reply Source/SessionManager/SessionManager.swift]: Thanks, done!!

**Review by billypchan — APPROVED** [review]

**Comment by marcomaddy1982** [comment]
@zenkins release major

**Pull request merged**